executor/cmd/executor/main.go

```diff
@@ -237,25 +240,40 @@ func main() {
 	grpcl := tcpm.MatchWithWriters(
 		cmux.HTTP2MatchHeaderFieldSendSettings("content-type", "application/grpc"))
 
-	clientRest, err := rest.NewJSONRestClient(*protocol, *sdepName, predictor, annotations)
-	if err != nil {
-		log.Fatalf("Failed to create http client: %v", err)
-	}
-	logger.Info("Running HTTP server ", "port", *port)
-	go runHttpServer(httpl, logger, predictor, clientRest, *port, false, serverUrl, *namespace, *protocol, *sdepName, *prometheusPath)
-
+	logger.Info("Running grpc server ", "port", *port)
 	var clientGrpc seldonclient.SeldonApiClient
 	if *protocol == "seldon" {
 		clientGrpc = seldon.NewSeldonGrpcClient(predictor, *sdepName, annotations)
 	} else {
 		clientGrpc = tensorflow.NewTensorflowGrpcClient(predictor, *sdepName, annotations)
 	}
-	logger.Info("Running gRPC server ", "port", *port)
 	go runGrpcServer(grpcl, logger, predictor, clientGrpc, *port, serverUrl, *namespace, *protocol, *sdepName, annotations)
 
+	clientRest, err := rest.NewJSONRestClient(*protocol, *sdepName, predictor, annotations)
+	if err != nil {
+		log.Fatalf("Failed to create http client: %v", err)
+	}
+	logger.Info("Running http server ", "port", *port)
+	go runHttpServer(httpl, logger, predictor, clientRest, *port, false, serverUrl, *namespace, *protocol, *sdepName, *prometheusPath)
+
+	if *serverType == "kafka" {
+		log.Info("Starting kafka server")
```

Comment:
May be worth exploring how to still expose some of the rest endpoints like `/metadata`